### PR TITLE
SITL: fix memory over-read detected by Valgrind

### PR DESCRIPTION
**Pull request by peterbarker**
==15803== Conditional jump or move depends on uninitialised value(s)
==15803==    at 0x4C34975: index (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==15803==    by 0x444D8D: SITL::Frame::load_frame_params(char const*) (SIM_Frame.cpp:363)
==15803==    by 0x445415: SITL::Frame::init(char const*, SITL::Battery*) (SIM_Frame.cpp:432)
==15803==    by 0x3696ED: SITL::MultiCopter::MultiCopter(char const*) (SIM_Multicopter.cpp:35)
==15803==    by 0x34B49C: SITL::MultiCopter::create(char const*) (SIM_Multicopter.h:44)
==15803==    by 0x34C58E: HALSITL::SITL_State::_parse_command_line(int, char* const*) (SITL_cmdline.cpp:480)
==15803==    by 0x344005: HALSITL::SITL_State::init(int, char* const*) (SITL_State.cpp:923)
==15803==    by 0x33D854: HAL_SITL::run(int, char* const*, AP_HAL::HAL::Callbacks*) const (HAL_SITL_Class.cpp:182)
==15803==    by 0x15ACDD: main (Copter.cpp:678)
==15803==

Scanning for `{` in an unterminated buffer.

### CODE REVIEW

File: libraries/SITL/SIM_Frame.cpp [hunk 1]
```diff
@@ -342,7 +345,8 @@ void Frame::load_frame_params(const char *model_json)
     if (fd == -1) {
         AP_HAL::panic("%s failed to load\n", model_json);
     }
-    char buf[st.st_size];
+    char buf[st.st_size+1];
```
Comment (khancyr) [inline libraries/SITL/SIM_Frame.cpp:348]:
```suggestion
    char buf[st.st_size+1] = {0}; 
```
maybe ?

Comment (peterbarker) [reply libraries/SITL/SIM_Frame.cpp:348]: clang++ was unhappy with the attempt to initialise this.

The memset will work :-)

**Review by tridge — APPROVED** [review]

File: libraries/SITL/SIM_Frame.cpp (hunk 1 above)
Comment (magicrub) [reply libraries/SITL/SIM_Frame.cpp:348]: We use the brackets everywhere else in the code.. why not be consistent?

**Comment by magicrub** [comment]
But it doesn't complain about the 100 other places we use it. Thus the confusion.

**Comment by peterbarker** [comment]
@magicrub interesting - can you link to a place in our code where we do initialise a dynamically-sized stack-stored array with braces?  Perhaps we're not crossing those codepaths in our valgrind runs?

Note that clang++ doesn't have a problem with `char foo[FOOSIZE] {}` - as that size is determined at compile-time.  It has a problem with `void bar(uint8_t foosize) { char foo[foosize] = {}; }` as foosize isn't compile-time determinable (probably).

**Pull request merged**